### PR TITLE
Ensure net http client code also works with the RoundTripper interface

### PR DESCRIPTION
**Pull request by grcevski**
With this PR I'm adding support for tracking HTTP client requests which use the RoundTripper interface. The client.Do probe will not catch these events, but there's a lower level rountrip method on Transport, which handles all HTTP client calls.

I'm also adding a new integration test which covers the now fixed issue for duplicate server requests and missing response codes (for https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/472), as well as this RoundTripper client call.

### CODE REVIEW

File: internal/pkg/instrumentation/bpf/net/http/client/probe.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func New(logger logr.Logger) probe.Probe {
 			},
 		},
 		Uprobes: map[string]probe.UprobeFunc[bpfObjects]{
-			"net/http.(*Client).do": uprobeDo,
+			"net/http.(*Transport).roundTrip": uprobeDo,
```
Comment (RonFed) [inline internal/pkg/instrumentation/bpf/net/http/client/probe.go:82]: Probably better to rename `uprobeDo` to something like `uprobeRoundTrip`.
In addition, I think it is worth renaming the probe name in the `bpf.probe.c` file

File: Makefile [hunk 2]
```diff
@@ -142,7 +142,8 @@ license-header-check:
 	           exit 1; \
 	   fi
 
-.PHONY: fixture-nethttp fixture-gin fixture-databasesql
+.PHONY: fixture-nethttp fixture-gin fixture-databasesql fixture-nethttp-custom
```
Comment (RonFed) [inline Makefile:145]: If adding a fixture here, it is worth adding also one in the github workflows in `kind.yml`

**Comment by grcevski** [comment]
Thanks @RonFed, I made the requested changes and fixed up the lint issues.

**Review by RonFed — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ OpenTelemetry Go Automatic Instrumentation adheres to [Semantic Versioning](http
 ### Added
 
 - Add `WithResourceAttributes` `InstrumentationOption` to configure `Instrumentation` to add additional resource attributes. ([#522](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/522))
+- Add HTTP client instrumentation for `Transport.roundTrip`. ([#529](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/529))
```
[the remark this answers is not in the document]
Comment (grcevski) [reply CHANGELOG.md]: Done, thanks!

**Review by pellared — APPROVED** [review]
